shared/src/commonMain/kotlin/com/vml/tutorial/plantshop/basket/presentation/BasketComponent.kt

```diff
@@ -0,0 +1,66 @@
+package com.vml.tutorial.plantshop.basket.presentation
+
+import com.arkivanov.decompose.ComponentContext
+import com.vml.tutorial.plantshop.MR
+import com.vml.tutorial.plantshop.basket.data.BasketRepository
+import com.vml.tutorial.plantshop.basket.presentation.components.BasketEvent
+import com.vml.tutorial.plantshop.core.presentation.UiText
+import com.vml.tutorial.plantshop.core.utils.componentCoroutineScope
+import com.vml.tutorial.plantshop.plants.data.PlantsRepository
+import com.vml.tutorial.plantshop.plants.domain.Plant
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.SharingStarted
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.combine
+import kotlinx.coroutines.flow.stateIn
+import kotlinx.coroutines.launch
+
+class BasketComponent(
+    componentContext: ComponentContext,
+    private val plantsRepository: PlantsRepository,
+    private val basketRepository: BasketRepository,
+    private val onNavigateToHome: () -> Unit,
+    private val onShowMessage: (message: UiText) -> Unit,
+): ComponentContext by componentContext {
+
+    private val plantsFlow: MutableStateFlow<List<Plant>> = MutableStateFlow(listOf())
+    val state: StateFlow<BasketScreenState> = plantsFlow.combine(basketRepository.basketItemsFlow) { plants, basket ->
+        BasketScreenState(
+            items = basket.mapNotNull { item ->
+                plants.firstOrNull { it.id == item.plantId }?.let { plant ->
+                    BasketItemState(plant, item.quantity)
+                }
+            }
+        )
+    }.stateIn(componentCoroutineScope(), SharingStarted.Lazily, BasketScreenState())
+
+    init {
+        componentCoroutineScope().launch {
+            plantsFlow.emit(plantsRepository.getPlants())
+            basketRepository.getBasketItems()
+        }
+    }
+
+    fun onEvent(event: BasketEvent) {
+        componentCoroutineScope().launch {
+            when(event) {
+                BasketEvent.Checkout -> checkout()
```

Comment:
checkout() already launches a new coroutine so, it might be best to move this function call outside of the launch block

Reply:
You are right :) nice